src/script/interpreter.cpp

```diff
@@ -2041,6 +2042,148 @@ bool GenericTransactionSignatureChecker<T>::CheckSequence(
 template class GenericTransactionSignatureChecker<CTransaction>;
 template class GenericTransactionSignatureChecker<CMutableTransaction>;
 
+bool VerifyScriptPostConditions(const std::vector<valtype> stack,
+                                const CScript &scriptSig, uint32_t flags,
+                                const ScriptExecutionMetrics &metrics,
+                                ScriptError *serror) {
+    // The CLEANSTACK check is only performed after potential P2SH evaluation,
+    // as the non-P2SH evaluation of a P2SH script will obviously not result in
+    // a clean stack (the P2SH inputs remain).
+    if ((flags & SCRIPT_VERIFY_CLEANSTACK) != 0) {
```

Comment:
We don't really need this for taproot execution, right? We could just check the stack size along with bool.